functions/src/lobby.ts

```diff
@@ -213,6 +223,89 @@ export const collectPromptAnswers = functions.firestore
     });
   });
 
+export async function addRole(lobbyDocRef: firestore.DocumentReference<Lobby>) {
+  const lobby = await lobbyDocRef.get();
+  if (lobby.exists === false) {
+    return { error: "Lobby doesn't exist!" };
+  }
+  const privatePlayerCollection = getPrivatePlayerCollection(lobbyDocRef);
+  const privatePlayers = (await privatePlayerCollection.get()).docs;
+
+  const { players, uids } = lobby.data() as Lobby;
+
+  for (let i = 0; i < players.length; i++) {
+    //  private player id = user uid
+    const privateId = privatePlayers[i].id;
+    // private player info
+    const privateInfo = privatePlayers[i].data();
+    // get player
+    const player = players[uids.indexOf(privateId)];
+    // replace the role of the player
+    player.role = privateInfo.role;
+  }
+  let catfishCount = 0;
+
+  for (let i = 0; i < uids.length; i++) {
+    // check the alive players
+    if (players[i].alive == true) {
+      if (players[i].role == "CATFISH") {
+        catfishCount++;
+      }
+    }
+    // check any catfish who are not alive and add their display name to the list
+  }
+
+  let winner: Role;
+  if (catfishCount == 0) {
+    winner = "CAT";
+  } else {
+    // if (catCount <= catfishCount) {
+    winner = "CATFISH";
+  }
+
+  return lobbyDocRef.update({ players, winner });
+}
+
+// when returning to the lobby, make every player alive again
+export async function makeAlive(lobbyDocRef: firestore.DocumentReference<Lobby>) {
+  const lobby = await lobbyDocRef.get();
+  if (lobby.exists === false) {
+    return { error: "Lobby doesn't exist!" };
+  }
+
+  // loop through each player and set their alive status to true
+  const { players } = lobby.data() as Lobby;
+  for (let i = 0; i < players.length; i++) {
+    players[i].alive = true;
+  }
+  return lobbyDocRef.update({ players });
+}
+
+export async function deletePrivatePlayers(
```

Comment:
it might be a good idea to querry instead of using uids from lobby but I'm not sure

Reply:
If it becomes an issue I'll look further into it